server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java

```diff
@@ -67,16 +79,34 @@ long getTotalBytesToUpload() {
     }
 
     @Override
-    public void onSuccess(TransferFileSnapshot fileSnapshot) {
+    public void onSuccess(TranslogCheckpointSnapshot fileSnapshot) {
         try {
             long durationInMillis = (System.nanoTime() - fileTransferStartTime) / 1_000_000L;
             remoteTranslogTransferTracker.addUploadTimeInMillis(durationInMillis);
-            remoteTranslogTransferTracker.addUploadBytesSucceeded(bytesForTlogCkpFileToUpload.get(fileSnapshot.getName()));
+            remoteTranslogTransferTracker.addUploadBytesSucceeded(bytesForTlogCkpFileToUpload.get(fileSnapshot.getTranslogFileName()));
+            remoteTranslogTransferTracker.addUploadBytesSucceeded(bytesForTlogCkpFileToUpload.get(fileSnapshot.getCheckpointFileName()));
         } catch (Exception ex) {
             logger.error("Failure to update translog upload success stats", ex);
         }
+        addGeneration(fileSnapshot.getGeneration(), TransferState.SUCCESS);
+        add(fileSnapshot.getTranslogFileName(), TransferState.SUCCESS);
+        if (!startUploadingTranslogCkpAsMetadata) {
```

Comment:
Shouldn't the statement - `add(fileSnapshot.getTranslogFileName(), TransferState.SUCCESS);` also come inside the if condition?

Reply:
No, this is what we are trying to achieve.
1. upload method without md.-> (mark generation, translog.tlog and translog.ckp as success)
2. upload method with md -> (mark generation, translog.tlog as success) we don't mark translog.ckp success in this case.

Reply:
Why do we need to track the translog status individually for case with md when generation tracking is one and same as translog?

Reply:
Yes, this will work with not marking the status of translog.tlog in case of with md. I was making sure that fileTransferTracker records information of file that are actually uploaded to remote. In addition to that marking this translog.tlog file as success also helps in UT's .

Reply:
Having clear segregation will help us to shift to a state where the classes are separate but with common parent.

Reply:
Making this change.